src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1630,6 +1619,10 @@ void TorrentHandle::handleMetadataReceivedAlert(libt::metadata_received_alert *p
         m_speedMonitor.reset();
         m_session->handleTorrentPaused(this);
     }
+    
+    // If first/last piece priority was specified when adding this torrent, set it up now that we can:
+    if (m_hasFirstLastPiecePriority)
+        setFirstLastPiecePriority(true);
```

Comment:
Not necessary. We can set sequential downloading on a torrent before we have its metadata. Not so for first / last piece priority.